tests/TestCase/Database/Type/DateTypeTest.php

```diff
@@ -220,6 +253,26 @@ public function testMarshal($value, $expected): void
         $this->assertEquals($expected, $result);
     }
 
+    /**
+     * test marshaling data with different timezone
+     *
+     * @dataProvider marshalWithTimezoneProvider
+     * @param mixed $value
+     * @param mixed $expected
+     */
+    public function testMarshalWithTimezone($value, $expected): void
+    {
+        $defaultTimezone = date_default_timezone_get();
+        date_default_timezone_set('Europe/Vienna');
```

Comment:
I think we need to test a specific time where the change in time zone due to incorrect conversion would change the date.

Reply:
I made sure that the test `testMarshalWithTimezone` is made to do exactly that.
With the old implementation of `return new FrozenDate('@' . $value->getTimestamp());` this will fail.

Reply:
Ok, I just didn't want it to accidentally test as we could change the sample dates in the provider without understanding the interaction.